### PR TITLE
Added Docker container linking

### PR DESCRIPTION
**Pull request by Fokko**
Added Docker container linking so that the user-container gets linked to the jupyterhub container, which makes communication much easier.

### CODE REVIEW

**Comment by rgbkrk** [comment]
This is fine with me, though I'll never use it as linking is on the table for deprecation in Docker (relying on docker networks and volumes).

**Comment by Fokko** [comment]
@rgbkrk thanks. I would like to see this merged into jupyterhub and upgrade to networks in the short future.

I've pushed a small update. I also link to a Spark container, but had some problems of overwriting keys in the dicts as `.update` does not work recursively.

File: dockerspawner/dockerspawner.py [hunk 1]
```diff
@@ -150,6 +150,18 @@ def client(self):
         )
     )
 
+    use_docker_links = Bool(
```
Comment (minrk) [inline dockerspawner/dockerspawner.py]: Shouldn't `links` be a dict, rather than a bool for hardcoded values?

Comment (Fokko) [inline dockerspawner/dockerspawner.py]: Definitely.

**Comment by rgbkrk** [comment]
Whoa, just noticed we're using PyFlakes on a Jupyter project. That's awesome.

**Comment by minrk** [comment]
@rgbkrk I added it as a band-aid while there are zero real tests.

**Pull request merged**